### DISCUSSION TITLE
Attach dlv into running milvus

### DISCUSSION THREAD

**Discussion by hyojeongyunn** [milvus-io/milvus#32758] [Q&A and General discussion]
Hello.
I'm trying to debug running milvus, but I have some problem with it.

Is it possible to attach dlv to running milvus? (I can run unittest with my vscode debugger.)

Whenever I try debug on running milvus, it always makes this error and stop. - "Encountered an unexpected DWARF CFA opcode: 0x2e" And the call stack of this error is from dlv, not milvus.
So, I cannot send query to it.

I'm debugging on vscode with below `launch.json` configurations.
```
"configurations": [
        {
            "name": "Launch Package",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "program": "/home/cmd/main.go",
            "args": ["run", "standalone"],
            "showLog": false,
            "env": {
                "LD_LIBRARY_PATH": "${LD_LIBRARY_PATH}:/home/lib",
                "PKG_CONFIG_PATH": "${PKG_CONFIG_PATH}:/home/internal/core/output/lib/pkgconfig"
            },
            "dlvFlags": ["--check-go-version=true"]
        }
]
```
cf. I cloned milvus repo onto my `home` directory so the path are like above.

For more information, 
I'm working on linux, ubuntu20.04.
I followed this page [build](https://github.com/milvus-io/milvus/tree/v2.4.1/build), and I'm running containers with this command `./scripts/devcontainer.sh up`.

My dlv version is 1.22.1 and go version is 1.20.7 linux/amd64.

**Comment by xiaofan-luan** [comment]
@SimFG any idea?

**Comment by yhmo** [comment]
Usually, debugging into Milvus doesn't help our work. There are many async threads running in Milvus, a debug breakpoint might cause timeout of other threads and terminate the entire process. Print log is a common way to address the issues.

**Reply by xiaofan-luan** [reply]
in fact, the dlv to helps a lot to check where milvus is stucked. I succesfully run dlv on milvus 2.2 but not recently.
On the otherside, pprof can help with the stack check

**Reply by hyojeongyunn** [reply]
First, thanks for the answer.
If it is the problem of multi-threads, one can try on gdb with multi-thread debugging option. [How to restrict gdb debugging to one thread at a time](https://stackoverflow.com/questions/6721940/how-to-restrict-gdb-debugging-to-one-thread-at-a-time)

I figured out that at least I could stop milvus(v2.4) at the given breakpoint with gdb. But, failed to continue the program. 😢

**Comment by SimFG** [comment]
@hyojeongyunn You can try to perform some actions,
1. reset some env var, (which is the `scripts/setenv.sh` file)
![image](https://github.com/milvus-io/milvus/assets/21985684/c289d2f3-8efc-45b9-8daf-001b4fe15947)
2. Include debugging information when compiling
To include debugging information during compilation, you need to modify the milvus command under the makefile file. Refer to the example to modify the go build command and add the `gcflags` parameter
```
go build -gcflags "all=-N -l" main.go
```
3. It may also be caused by environmental problems， refer to: https://stackoverflow.com/questions/608361/unhandled-dwarf-expression
